sideload.go

```diff
@@ -0,0 +1,420 @@
+package neutrino
+
+import (
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/chaindataloader"
+	"github.com/lightninglabs/neutrino/chainsync"
+	"github.com/lightninglabs/neutrino/headerfs"
+	"github.com/lightninglabs/neutrino/headerlist"
+)
+
+const (
+	sideloadDataChunkSize int32 = 2000
+)
+
+// sideloader a struct that contains all dependencies required to fetch and
+// process sideloaded headers.
+type sideloader struct {
+	// reader is the sideload source.
+	blkHdrReader chaindataloader.Reader
+
+	cfHdrReader chaindataloader.Reader
+	// blkHdrProcessor contains dependies required to verify and store block
+	// headers.
+	blkHdrProcessor *blkHdrProcessor
+
+	// blkHdrSkipVerify indicates if processing the headers require verification.
+	blkHdrSkipVerify bool
+
+	fHdrSkipVerify bool
+
+	// fHdrNextCheckpointIdx indicates the index of the next filter header
+	// checkpoint height relative to the current filter tip.
+	fHdrNextCheckpointIdx int
```

Comment:
This information can be encapsulated in the the interface I mentioned earlier. This leaks a lot of details, when the main actions here are: parse next headers, validate them, write them to disk.